lib/compat/wordpress-6.3/class-gutenberg-rest-blocks-controller.php

```diff
@@ -0,0 +1,47 @@
+<?php
+/**
+ * Reusable blocks REST API: WP_REST_Blocks_Controller class
+ *
+ * @package WordPress
+ * @subpackage REST_API
+ * @since 5.0.0
+ */
+
+/**
+ * Controller which provides a REST endpoint for the editor to read, create,
+ * edit and delete reusable blocks. Blocks are stored as posts with the wp_block
+ * post type.
+ *
+ * @since 5.0.0
+ *
+ * @see WP_REST_Posts_Controller
+ * @see WP_REST_Controller
+ */
+class Gutenberg_REST_Blocks_Controller extends WP_REST_Blocks_Controller {
+	/**
+	 * Filters a response based on the context defined in the schema.
+	 *
+	 * @since 5.0.0
+	 * @since 6.3 Adds the `wp_pattern_sync_status` property to the response.
+	 *
+	 * @param array  $data    Response data to filter.
+	 * @param string $context Context defined in the schema.
+	 * @return array Filtered response.
+	 */
+	public function filter_response_by_context( $data, $context ) {
+		$data = parent::filter_response_by_context( $data, $context );
+
+		/*
+		 * Remove `title.rendered` and `content.rendered` from the response. It
+		 * doesn't make sense for a reusable block to have rendered content on its
+		 * own, since rendering a block requires it to be inside a post or a page.
+		 */
+		unset( $data['title']['rendered'] );
+		unset( $data['content']['rendered'] );
+
+		// Add the core wp_pattern_sync_status meta as top level property to the response.
+		$data['wp_pattern_sync_status'] = $data['meta']['wp_pattern_sync_status'];
+		unset( $data['meta']['wp_pattern_sync_status'] );
```

Comment:
Do we need to make sure that we still have `$data['meta']['wp_pattern_sync_status']` set after filters are applied, before attempting to access it?

Reply:
Yeh, anybody's guess what other filters might do, have added an isset check.

Reply:
We'll need to make this same change on the [backport](https://github.com/WordPress/wordpress-develop/pull/4761/files#diff-db0cd20d54a532e62f13adcd1fbb627cb28dfec58410af70feb2cb6dcbd86452R60-R62) as well.